build-support/source_roots/source_roots.json

```diff
@@ -0,0 +1,121 @@
+[
```

Comment:
As discussed in slack, I think that this is a non-starter for larger repos.

Reply:
To clarify, the existing pattern system is also a non-starter for larger repos, unless they use some detectable pattern like `src/python` under every project, which is not idiomatic in Python projects.  We can still allow patterns, but they are not going to solve the problem.

Reply:
A follow up to this change will need to support some sort of sentinel file mechanism, which is what we had back in the distant past, I believe...